docs/articles/concepts/kafka.md

```diff
@@ -0,0 +1,167 @@
+Bytewax ships with connectors for [Kafka](https://www.confluent.io/) and
+Kafka compatible systems, like [Redpanda](https://redpanda.com/).
+
+In this section, we'll discuss these connectors in detail, as well as
+provide some important operational information about using them as sources
+and sinks.
+
+## Connecting to Kafka
+
+Bytewax provides two basic ways to connect to Kafka.
+
+You can use [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource)
+and [`KafkaSink`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSink) directly:
+
+```python
+from bytewax.connectors.kafka import KafkaSource, KafkaSink, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = op.input("kafka-in", flow, KafkaSource(brokers, ["in-topic"]))
+processed = op.map("map", kinp, lambda x: KafkaSinkMessage(x.key, x.value))
+op.output("kafka-out", processed, KafkaSink(brokers, "out-topic"))
+```
+
+Or use the Kafka input [operator](/apidocs/bytewax.connectors/kafka/operators#bytewax.connectors.kafka.operators.input):
+
+```python
+from bytewax.connectors.kafka import operators as kop, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = kop.input("kafka-in", flow, brokers=brokers, topics=["in-topic"])
+processed = op.map("map", kinp.oks, lambda x: KafkaSinkMessage(x.key, x.value))
+kop.output("kafka-out", processed, brokers=brokers, topic="out-topic")
+```
+
+## Error handling
+
+The [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource) input
+source will raise an exception whenever an error is encountered when consuming from Kafka.
+
+The Kafka input [operator](/apidocs/bytewax.connectors/kafka/operators#bytewax.connectors.kafka.operators.input)
+returns a dataclass containing two output streams. The `.oks` field is a stream of
+[`KafkaMessage`](/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSourceMessage)
+that were successfully processed. The `.errs` field is a stream of [`KafkaError`](/apidocs/bytewax.connectors/kafka/error#bytewax.connectors.kafka.message.KafkaError)
+messages where an error was encountered. Items that encountered an error have their `.err` field set with more
+details about the error.
+
+Note that if no processing is attached to the `.errs` stream of messages, they will be silently
+dropped, and processing will continue.
+
+Alternatively, error messages in the `.errs` stream can be published to a "dead letter queue" where they
```

Comment:
short clarification of what we mean by that (separate Kafka topic for bad messages) or, ideally, a very short code snippet how one does that would be amazing to have here

Reply:
something like this?

```python
errs = op.map(
    "kafka-errors",
    kinp.errs,
    lambda res: KafkaSinkMessage(
        key=None, value=json.dumps({"error": f"{res.err}", "msg": res.msg.value})
    ),
)
kop.output("dlq", errs, topic="errors", brokers=BROKERS)

```

That also makes me think that we should make the `KafkaError` (and so also `KafkaSourceMessage`) easily serializable, so that all the metadata can be sent